packages/devtools_app/lib/src/trees.dart

```diff
@@ -234,6 +234,29 @@ class TreeNode<T extends TreeNode<T>> {
     }
     return currentNode;
   }
+
+  TreeNode<T> shallowCopy();
+
+  /// Filters a tree starting at this node and returns a list of new roots after
+  /// filtering.
+  ///
+  /// If the root [this] should be included in the filtered results, the list
+  /// will contain one node. If the root [this] should not be included in the
+  /// filtered results, the list may contain one or more nodes.
```

Comment:
it isn't clear whether the filter returning true means the node should be filtered out or retained.